### PR TITLE
FIX-#2320: Raise exceptions in read_csv in some cases with `skipfooter!=0`

### LINKED ISSUE #2320: read_csv with Ray engine doesn't raise exceptions while Pandas raises
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Modin version** (`modin.__version__`): 0.8.1.1+34.ga571e10
- **Python version**: 3.8.6
- **Code we can use to reproduce**:
```
import os

os.environ["MODIN_ENGINE"] = "ray"

import pandas
import modin.pandas as pd
from modin.pandas.test.utils import df_equals

test_filename = "test.csv"
kwargs = {
    "filepath_or_buffer": test_filename,
    "skipfooter": 1,
    "nrows": 1,
}
str_two_cols = """col1,col2
one,two
0,1
2,3
"""

try:
    with open(test_filename, "w") as f:
        f.write(str_two_cols)

    try:
        df_pandas = pandas.read_csv(**kwargs)
    except Exception as exc:
        print("pandas.read_csv raised exception:\n", exc)
    df_pd = pd.read_csv(**kwargs)
    print("pd.read_csv output:\n", df_pd)
finally:
    os.remove(test_filename)
```

### Source code / logs

```
pandas.read_csv raised exception:
 'skipfooter' not supported with 'nrows'
pd.read_csv output:
 Empty DataFrame
Columns: [col1, col2]
Index: []
```

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2320 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -335,7 +335,6 @@ def test_read_csv_parsing_1(
 
         eval_io(
             fn_name="read_csv",
-            check_exception_type=None,  # issue #2320
```
Comment (YarShev) [inline modin/pandas/test/test_io.py:338]: Why don't we want to check exception type anymore?

File: modin/core/io/text/text_file_dispatcher.py [hunk 2]
```diff
@@ -661,6 +661,10 @@ def check_parameters_support(
         if read_kwargs["lineterminator"] is not None:
             return (False, "`lineterminator` parameter is not supported")
 
+        if read_kwargs.get("skipfooter"):
+            if read_kwargs.get("nrows") or read_kwargs.get("engine") == "c":
+                return (False, "raise exception by pandas itself")
```
Comment (YarShev) [inline modin/core/io/text/text_file_dispatcher.py:666]:
```suggestion
                return (False, "Exception is raised by pandas itself")
```

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py [hunk 3]
```diff
@@ -454,6 +454,9 @@ def _read_csv_check_support(
                     f"Invalid file path or buffer object type: {type(filepath_or_buffer)}"
                 )
 
+        if read_csv_kwargs.get("skipfooter") and read_csv_kwargs.get("nrows"):
+            return (False, "raise exception by pandas itself")
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py:458]:
```suggestion
            return (False, "Exception is raised by pandas itself")
```

**Comment by YarShev** [comment]
Do the changes fix both the cases mentioned in the issue #2320?

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py:338]: On the contrary, now we are checking. `None` was used to disable validation.

**Comment by anmyachev** [comment]
> Do the changes fix both the cases mentioned in the issue #2320?

No, create a separate issue https://github.com/modin-project/modin/issues/5526.

**Review by dchigarev — COMMENTED** [review]
Why do we prefer falling back to pandas (by claiming the parameters set as unsupported) instead of explicitly raising a proper exception from Modin?

**Comment by anmyachev** [comment]
> Why do we prefer falling back to pandas (by claiming the parameters set as unsupported) instead of explicitly raising a proper exception from Modin?

This hides one of the implementation details - the error message itself.

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (YarShev) [reply modin/pandas/test/test_io.py:338]: Sorry for confusion 😄

**Review by YarShev — APPROVED** [review]
LGTM! @dchigarev, do you have any other comments?

**Review by dchigarev — APPROVED** [review]

**Pull request merged**